### PR TITLE
Update the documentation for 0.6.6 and announce it.

### PR DESCRIPTION
**Pull request by sjrd**
Scaladoc for the library of 0.6.6 available at http://www.scala-js.org/api/scalajs-library/0.6.6/#scala.scalajs.js.package

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @gzm0, @nicolasstucki and anyone else who wants to.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    new js.ConstructorTag[C](js.constructorOf[js.Array[Int]]))
+{% endhighlight %}
+
+although you cannot write the desugared version in user code because the constructor of `js.ConstructorTag` is private.
+
+This feature is particularly useful for Scala.js libraries wrapping JavaScript frameworks expecting to receive JavaScript constructors as parameters.
+
+### Query for Development versus Production mode
+
+A pair of methods allow you to query in your code whether the developer is doing a Development build or a Production build (for all practical purposes, aka `fastOptJS` and `fullOptJS`, resp.): `isDevelopmentMode` and `isProductionMode` in the object [`scala.scalajs.LinkingInfo`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.LinkingInfo$).
```
Comment (gzm0) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: This reads a bit hard. How about:

"The new methods `isDevelopmentMode` and `isProductionMode` allow you to query in your code whether you are running in a Development build or a Production build (for all ...). The methods are located in [`scala.scalajs.LinkingInfo`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.LinkingInfo$)."

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+## Release notes
+
+As a minor release, 0.6.6 is (almost, see Breaking changes below) backward source and binary compatible with previous releases in the 0.6.x series.
+Libraries compiled with earlier versions can be used with 0.6.6 without change.
+However, it is not forward compatible: libraries compiled with 0.6.6 cannot be used by projects using 0.6.{0-5}.
+
+Please report any issues [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Breaking changes
```
Comment (gzm0) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: Maybe add to this section that anything in `org.scalajs.tools` might have experienced a breaking change.

Comment (sjrd) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: OK

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md (hunk 1 above)
Comment (sjrd) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: OK

**Comment by sjrd** [comment]
Updated.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+As of Scala.js 0.6.6, this setting is redundant, as it is the default, but `run` and `test` will still use Rhino by default!
+To disable Rhino and use Node.js/PhantomJS in 0.6.6, use the following setting instead:
+
+    scalaJSUseRhino in Global := false
+
+### Tools API
+
+The [Tools API]({{ site.production_url }}/api/scalajs-tools/0.6.6/#org.scalajs.core.tools.package),
+(the API to the Scala.js linker and optimizer) have been completely redesigned.
+Users of this API, such as Scala.jsFiddle, will experience breakages.
```
Comment (gzm0) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: Is that really how one writes fiddle?

Comment (nicolasstucki) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: I believe it is so, but I would separate the fiddle from the js with a space or a dash.

Comment (gzm0) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: The title of fiddle says "Scala-Js-Fiddle"...

**Comment by gzm0** [comment]
LGTM otherwise

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+For example:
+
+    val scalaT: (Int, String) = t
+    val t2: js.Tuple2[Int, String] = scalaT
+    val js.Tuple2(a, b) = t2
+
+### Support for JUnit
+
+It is now possible to write JUnit tests for Scala.js!
+We have ported the entire JUnit API to Scala.js, so that it is now possible to write cross-compiling unit tests with JUnit.
```
Comment (nicolasstucki) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: I would remove the word "entire". I think `We have ported the JUnit API to Scala.js, ...` should be good.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+To enable JUnit on your Scala.js project, add the appropriate sbt AutoPlugin using
+
+    enablePlugins(ScalaJSJUnitPlugin)
+
+(like you would with the basic `ScalaJSPlugin`).
+For a `crossProject`, this must be enabled only on the JS variant.
```
Comment (nicolasstucki) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: Should also mention that on the JVM variant the normal JUnit dependency should be added.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md (hunk 3 above)
Comment (sjrd) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: Its HTML title says that, but the landing page says "Scala.jsFiddle": http://www.scala-js-fiddle.com/gist/9443f8e0ecc68d1058ad/LandingPage.scala

**Comment by nicolasstucki** [comment]
That is all, otherwise LGTM.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md (hunk 4 above)
Comment (sjrd) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: OK.

**Comment by sjrd** [comment]
Updated for the "entire". In doubt, I haven't changed the fiddle name.

File: _posts/news/2016-01-25-announcing-scalajs-0.6.6.md (hunk 3 above)
Comment (sjrd) [inline _posts/news/2016-01-25-announcing-scalajs-0.6.6.md]: I made it into a link, so at least it doesn't have the "is there a dot/comma/space?" problem

**Pull request merged**